crates/matrix-sdk-base/src/client.rs

```diff
@@ -705,64 +710,78 @@ impl BaseClient {
                     continue;
                 }
             };
+            account_data.insert(event.event_type(), raw_event.clone());
+            parsed_events.push(event);
+        }
 
-            if let AnyGlobalAccountDataEvent::Direct(e) = &event {
-                let mut new_dms = HashMap::<&RoomId, HashSet<OwnedUserId>>::new();
-                for (user_id, rooms) in e.content.iter() {
-                    for room_id in rooms {
-                        new_dms.entry(room_id).or_default().insert(user_id.clone());
-                    }
+        changes.account_data = account_data;
+        parsed_events
+    }
+
+    /// Processes the direct rooms in a sync response:
+    ///
+    /// Given a [`StateChanges`] instance, processes any direct room info
+    /// from the global account data and adds it to the room infos to
+    /// save.
+    #[instrument(skip_all)]
+    pub(crate) async fn process_direct_rooms(
+        &self,
+        events: Vec<AnyGlobalAccountDataEvent>,
```

Comment:
```suggestion
        events: &[AnyGlobalAccountDataEvent],
```

A `Vec` demands an allocation, but in this code, we may not need one. So requiring a slice is better here. Read my other comments to see how it translates.